src/highlight/HighlightAnnotations.tsx

```diff
@@ -118,6 +93,21 @@ const HighlightAnnotations = (props: Props): JSX.Element => {
         stageSelection();
     }, [isCreating, selection]); // eslint-disable-line react-hooks/exhaustive-deps
 
+    React.useEffect(() => {
+        if (highlightRef === null || !staged) {
+            return;
+        }
+
+        const { height, width, top, left } = highlightRef.getBoundingClientRect();
+
+        setReferenceShape({
+            height,
+            width,
+            x: left,
+            y: top,
+        });
```

Comment:
We need to do the destructuring and rename to `x, y` in every type annotations (highlight, region, drawing). How about moving this logic in `creator/actions.ts`? Then we only need to call it like `setReferenceShape(highlightRef.getBoundingClientRect())`

Reply:
Good idea